backend/src/main/java/com/ksidelta/calendal/main/domain/source/Source.java

```diff
@@ -0,0 +1,33 @@
+package com.ksidelta.calendal.main.domain.source;
+
+import com.ksidelta.calendal.main.domain.Event;
+
+import java.util.List;
+
+public abstract class Source {
+    String id;
+    String sourceName;
+
+    public Source(String id, String sourceName) {
+        this.id = id;
+        this.sourceName = sourceName;
+    }
+
+    public abstract List<Event> getEvents();
```

Comment:
This can be a little tricky to implement architecture wise, ask me for help if you don't have idea how to do it properly.